builtin/logical/pki/crl_util.go

```diff
@@ -367,7 +367,32 @@ func (cb *crlBuilder) getPresentLocalDeltaWALForClearing(sc *storageContext) ([]
 }
 
 func (cb *crlBuilder) getPresentUnifiedDeltaWALForClearing(sc *storageContext) ([]string, error) {
-	return cb._getPresentDeltaWALForClearing(sc, unifiedDeltaWALPath)
+	walClusters, err := sc.Storage.List(sc.Context, unifiedDeltaWALPrefix)
+	if err != nil {
+		return nil, fmt.Errorf("error fetching list of clusters with delta WAL entries: %w", err)
+	}
+
+	var allPaths []string
+	for index, cluster := range walClusters {
+		if cluster[len(cluster)-1] != '/' {
```

Comment:
hmm, if cluster doesn't end with a '/' doesn't that mean we are a file and not a directory? So we should be skipping this entry not appending a '/' to it?

Reply:
Hmm, that's probably better. I think in other places we do that, `path_revoke.go` for instance. In others we just assume the `/` so I'm fine assuming the `/`, that's more consistent.